OrbitCore/LinuxPerfEventProcessor.h

```diff
@@ -0,0 +1,79 @@
+//-----------------------------------
+// Copyright Pierric Gimmig 2013-2019
+//-----------------------------------
+#pragma once
+
+#include <queue>
+#include "LinuxPerfEvent.h"
+#include "LinuxPerfEventVisitor.h"
+
+// A comparator used for the priority queue, such that pop/top will
+// always return the oldest event in the queue.
+class TimeStampCompare
+{
+public:
+    bool operator() (std::shared_ptr<LinuxPerfEvent> lhs, std::shared_ptr<LinuxPerfEvent> rhs)
+    {
+        return lhs->Timestamp() > rhs->Timestamp();
+    }
+};
+
+class LinuxPerfEventProcessor
+{
+public:
+    // While processing, we do not touch the events with a timestamp less 
+    // then 1/100 sec smaller than the most recent one in the queue.
+    // This way we can ensure, that all events (from different sources)
+    // are processed in the correct order.
+    uint64_t OFFSET = 10000000 /*ns*/;
```

Comment:
Consider adding the unit "ns" to the OFFSET name, like OFFSET_IN_NS, or something like that.